src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+﻿using System;
+using System.Reactive.Disposables;
+using System.Windows.Input;
+using Avalonia.Controls.Metadata;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.LogicalTree;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// A button with primary and secondary parts that can each be pressed separately.
+    /// The primary part behaves like a <see cref="Button"/> and the secondary part opens a flyout.
+    /// </summary>
+    [PseudoClasses(
+        ":disabled",
+        ":secondaryButtonRight",
+        ":secondaryButtonSpan",
+        ":checkedFlyoutOpen",
+        ":flyoutOpen",
+        ":checkedTouchPressed",
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
```

Comment:
As an example, you definitely don't need "touchPressed" and "checkedTouchPressed" pseudoclasses, when you already have default "pressed" and "checked", and you can add single one "touch". After that it can be combined as ":touch:pressed:checked".

Reply:
See above:

 1. I'm sure this is fine for the first version of the control, it matches with other implementations.
 2. You are making it extremely difficult to port upstream WinUI Fluent style changes in the future. Well, at least a lot more thought will need to be done and a full understanding of the style is needed.

Reply:
We are already making so many changes in the Style, I think maintaining easy compatibility with upstream Fluent / Fluentv2 changes doesn't matter now. So that decision seems to have been made -- it just requires much more work for future syncronizations.

Reply:
Was changed in later commits so resolved. Discussion continues here: 

https://github.com/AvaloniaUI/Avalonia/pull/7422/files#r792961686